### PR TITLE
Quest log

### PR DESCRIPTION
**Pull request by wushin**

### CODE REVIEW

File: src/map/script-fun.cpp [hunk 1]
```diff
@@ -496,6 +496,67 @@ void builtin_if (ScriptState *st)
...
+        OMATCH_BEGIN_SOME(quest_data, quest_data_)
+        {
+            VarName reg = quest_data->quest_vr;
+            int quest_shift = quest_data->quest_shift;
+            int quest_mask = quest_data->quest_mask;
+            for (i = 0; i < sd->status.global_reg_num; i++)
+            {
+                if (sd->status.global_reg[i].str == reg)
+                {
+                    val = ((sd->status.global_reg[i].value & quest_mask) >> quest_shift);
```
Comment (wushin) [inline src/map/script-fun.cpp]: @mekolat ^ then does this with those DB vars. Ought to look familiar.

File: src/map/pc.cpp [hunk 2]
```diff
@@ -3839,16 +3841,36 @@ int pc_readglobalreg(dumb_ptr<map_session_data> sd, VarName reg)
...
+    QuestId questid;
+    assert (sd->status.global_reg_num < GLOBAL_REG_NUM);
+    for (QuestId q = wrap<QuestId>(0); q < wrap<QuestId>(-1); q = next(q))
+    {
+        P<struct quest_data> quest_data_ = TRY_UNWRAP(questdb_exists(q), continue);
+        for (i = 0; i < sd->status.global_reg_num; i++)
+        {
+            if (sd->status.global_reg[i].str == quest_data_->quest_vr)
+            {
+                int val = ((sd->status.global_reg[i].value & quest_data_->quest_mask) >> quest_data_->quest_shift);
```
Comment (wushin) [inline src/map/pc.cpp]: @mekolat and this in the right area depending on the var.

File: src/map/script-fun.cpp [hunk 3]
```diff
@@ -496,6 +496,67 @@ void builtin_if (ScriptState *st)
...
+    ZString name = variable_names.outtern(reg.base());
+    char prefix = name.front();
+    char postfix = name.back();
+
+    if (prefix != '$')
+        sd = script_rid2sd(st);
+
+    if (postfix == '$')
+    {
+        // 文字列
```
Comment (mekolat) [inline src/map/script-fun.cpp]: you could've AT LEAST remove those japanese comments from your builtin

Comment (wushin) [inline src/map/script-fun.cpp]: I said it still needed polishing but the damn thing works.

Comment (mekolat) [inline src/map/script-fun.cpp]: :imp: DO IT NAO

Comment (wushin) [inline src/map/script-fun.cpp]: I am.

**Comment by wushin** [comment]
@mekolat now it uses getq is used set @sample, getq("VARNAME");

**Comment by wushin** [comment]
@mekolat ok, it now works that set and get in the default TMWA pick up on the sudo vars with no helpers. I'd suggest we should prefix them all QL_ or it will come back to bite us.

**Pull request merged**